slasher/rpc/detect_update_min_max_span.go

```diff
@@ -0,0 +1,103 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+type detect func(span uint64, spans *ethpb.MinMaxSpan, source uint64) uint64
+
+func detectMax(span uint64, spans *ethpb.MinMaxSpan, source uint64) uint64 {
+	maxSpan := uint64(spans.MaxSpan)
+	if maxSpan > span {
+		return maxSpan + source
+	}
+	return 0
+}
+
+func detectMin(span uint64, spans *ethpb.MinMaxSpan, source uint64) uint64 {
+	minSpan := uint64(spans.MinSpan)
+	if minSpan < span {
+		return minSpan + source
+	}
+	return 0
+}
+
+func (ss *Server) detectSpan(source, target, validatorIdx uint64, detectionFunc detect) (uint64, uint64, *ethpb.EpochSpanMap, error) {
```

Comment:
This func could really benefit from some basic comments even if it is unexported, maybe explaining that how it takes in a callback that does detection.

Reply:
adding